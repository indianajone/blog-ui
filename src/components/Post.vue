<template>
    <div class="section is-flex is-verticle is-block-center">
        <router-link class="button ml-auto is-primary is-outlined" :to="'/edit/' + post._id">
            Edit
        </router-link>
        <component :is="view" :post="post"></component>
    </div>
</template>

<script>
import TextView from '@/components/views/TextView';
import ImageView from '@/components/views/ImageView';

export default {
    name: 'Post',
    components: { TextView, ImageView },
    data() {
        return {
            post: {},
            view: null
        };
    },
    mounted() {
        this.fetch().then(() => {
            this.loadView();
        });
    },
    methods: {
        async fetch() {
            const { id } = this.$route.params;
            const { data } = await this.$http.get(`/posts/${id}`);
            this.post = data.data;
        },
        loadView() {
            this.view = `${this.titleCase(this.post.type)}View`;
        },
        titleCase(str) {
            return str.charAt(0).toUpperCase() + str.slice(1);
        }
    }
};
</script>

<style lang="scss">
    .post {
        margin: 1em;
    }
</style>

