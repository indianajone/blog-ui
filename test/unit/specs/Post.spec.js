import Post from '@/components/Post';
import { mount } from 'vue-test-utils';
import { createRenderer } from 'vue-server-renderer';

describe('Post.vue', () => {

    let wrapper;

    beforeEach(() => {
        wrapper = mount(Post, {
            mocks: {
                $route: {
                    params: { id: '59eccf0e4e8b8f72705ac87d' }
                },
                $http: {
                    defaults: { baseURL: 'localhost' },
                    get() {
                        return { data: {
                            data: {
                                title: 'New Post',
                                body: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit.',
                                _id: '59eccf0e4e8b8f72705ac87d',
                                createdAt: '2017-10-02T17:02:06.051Z',
                                type: 'text'
                            }
                        } };
                    }
                }
            }
        });
    });

    it('should matches snapshot', () => {
        const renderer = createRenderer();
        renderer.renderToString(wrapper.vm, (err, str) => {

            if (err) {
                throw new Error(err);
            }
            expect(str).toMatchSnapshot();

        });
    });

});
