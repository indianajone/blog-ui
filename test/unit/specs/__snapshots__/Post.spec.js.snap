// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post.vue should matches snapshot 1`] = `
<div class="section is-flex is-verticle is-block-center">
    <router-link to="/edit/59eccf0e4e8b8f72705ac87d" class="button ml-auto is-primary is-outlined">
        Edit
    </router-link>
    <div class="post">
        <h1 class="title">
            New Post
            <span class="subtitle">2017-10-02T17:02:06.051Z</span></h1>
        <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit.</p>
    </div>
</div>
`;
